expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/extension/SDKCoreResponseExtension.kt

```diff
@@ -0,0 +1,61 @@
+package com.expediagroup.sdk.rest.extension
+
+import java.io.IOException
+
+import com.expediagroup.sdk.core.http.Response as SDKCoreResponse
+import com.expediagroup.sdk.rest.model.Response
+import com.expediagroup.sdk.rest.trait.operation.OperationNoResponseBodyTrait
+import com.expediagroup.sdk.rest.trait.operation.OperationResponseBodyTrait
+import com.expediagroup.sdk.rest.trait.serialization.DeserializeResponseBodyTrait
+
+/**
+ * Extension function to parse the body of an SDKCoreResponse as a specific type.
+ *
+ * @param T the type of the response body
+ * @param operation the operation that defines the response body type
+ * @param deserializer the deserializer to use for parsing the response body
+ * @return the parsed response body of type T
+ */
+@Throws(IOException::class)
+internal fun <T> SDKCoreResponse.parseBodyAs(
+    operation: OperationResponseBodyTrait<T>,
+    deserializer: DeserializeResponseBodyTrait
+): T {
+    require(body != null) { "Response body is null!" }
```

Comment:
Make T not nullable in OperationResponseBodyTrait instead?

Reply:
https://github.com/ExpediaGroup/lodging-connectivity-java-sdk/pull/166/files#r1938447698